src/app/portlets/dot-edit-page/content/components/dot-edit-page-view-as-controller/dot-edit-page-view-as-controller.component.ts

```diff
@@ -0,0 +1,101 @@
+import { DotRenderedPageState, DotPageMode } from '@portlets/dot-edit-page/shared/models';
+
+@Component({
+    selector: 'dot-edit-page-view-as-controller',
+    templateUrl: './dot-edit-page-view-as-controller.component.html',
+    styleUrls: ['./dot-edit-page-view-as-controller.component.scss']
+})
+export class DotEditPageViewAsControllerComponent implements OnInit {
+    @Input() pageState: DotRenderedPageState;
+
+    isEnterpriseLicense$: Observable<boolean>;
+    messages: { [key: string]: string } = {};
+    previewDeviceLabel$: Observable<string>;
+
+    constructor(
+        private dotAlertConfirmService: DotAlertConfirmService,
+        private dotMessageService: DotMessageService,
+        private dotLicenseService: DotLicenseService,
+        private dotPageStateService: DotPageStateService,
+        private dotPersonalizeService: DotPersonalizeService
+    ) {}
+
+    ngOnInit(): void {
+        this.isEnterpriseLicense$ = this.dotLicenseService.isEnterprise();
+        this.previewDeviceLabel$ = this.dotMessageService
+            .getMessages(['editpage.viewas.previewing'])
+            .pipe(
+                take(1),
+                map((messages: { [key: string]: string }) => messages['editpage.viewas.previewing'])
+            );
+    }
+
+    /**
+     * Handle the changes in Persona Selector.
+     *
+     * @param DotPersona persona
+     * @memberof DotEditPageViewAsControllerComponent
+     */
+    changePersonaHandler(persona: DotPersona): void {
+        this.dotPageStateService.setPersona(persona);
+    }
+
+    /**
+     * Handle changes in Language Selector.
+     *
+     * @param DotLanguage language
+     * @memberof DotEditPageViewAsControllerComponent
+     */
+    changeLanguageHandler({ id }: DotLanguage): void {
+        this.dotPageStateService.setLanguage(id);
+    }
+
+    /**
+     * Handle changes in Device Selector.
+     *
+     * @param DotDevice device
+     * @memberof DotEditPageViewAsControllerComponent
+     */
+    changeDeviceHandler(device: DotDevice): void {
+        this.dotPageStateService.setDevice(device);
+    }
+
+    /**
+     * Remove personalization for the current page and set the new state to the page
+     *
+     * @param {DotPersona} { keyTag }
+     * @memberof DotEditPageViewAsControllerComponent
+     */
+    deletePersonalization({ keyTag }: DotPersona): void {
+        this.dotAlertConfirmService.confirm({
+            header: 'Delete personalization',
+            message: `Are you sure you want delete personalization for <b>${
+                this.pageState.viewAs.persona.name
+            }</b>? This action cannot be undone.`,
+            accept: () => {
+                this.dotPersonalizeService
+                    .despersonalized(this.pageState.page.identifier, keyTag)
+                    .pipe(take(1))
+                    .subscribe(() => {
+                        this.dotPageStateService.setLock(
```

Comment:
I don't know if setLock is a good name for this method, maybe we should have lock and unlock method and remoce the boolean parameter, maybe the lock method can go automatically to EDIT_MODE, and the unlock can go to PREVIEW_MODE

Reply:
But that's not how it works because you could also `lock` a page in EDIT and LIVE modes.

I know set lock is like we're always locking the page by calling this method and this is not the case it depends if boolean is true (to lock) or false (to unlock) I tried to find a better name but I couldn't think of anything but maybe we can rename to `setLockState`?